### ISSUE TITLE
Bokeh layout engine measurements should wait on external components to be fully loaded

### ISSUE THREAD

**Issue by MarcSkovMadsen** [bokeh/bokeh#10033]
This issues is based on https://github.com/holoviz/panel/issues/835 for Panel.

"We should probably file an issue with bokeh to ensure that measurements wait on external components to be fully loaded.". @philippjfr 

-------

#### System Info

- Panel: 0.7.0
- Bokeh: 1.4.0
- Python: 3.7.4
- OS: Windows 8.1
- Browser: Chrome

#### Description of expected behavior and the observed behavior

If I do a simple layout using columns and markdown with an image I would expect it to display nicely. 

It does not. Instead it displays on top of each other.

![image](https://user-images.githubusercontent.com/42288570/70017590-d106bc00-1583-11ea-9dfb-8d5b5f3f4399.png)

#### Complete, minimal, self-contained example code that reproduces the issue

```python
import panel as pn

ABOUT = """\
# About

[Panel](https://panel.pyviz.org/) is a powerfull framework for building awesome-analytics apps in [Python](https://www.python.org/).

The purpose of this app is to test that a **multi-page Dashboard Layout** similar to the [bootstrap dashboard template](https://getbootstrap.com/docs/4.3/examples/dashboard/) from [getboostrap.com](https://getbootstrap.com/) can be implemented in [Panel](https://panel.pyviz.org/).

[![GetBootstrap Dashboard Template](https://getbootstrap.com/docs/4.4/assets/img/examples/dashboard.png)](https://getbootstrap.com/docs/4.3/examples/dashboard/)
"""

INFO = """\
Navigate to the **Dashboard Page** via the **Sidebar** to see the result.
Or Navigate to the **Limitations Page** to learn of some of the limitations of Panel that
I've experienced."""

about = pn.pane.Markdown(ABOUT)
info = pn.pane.Markdown(INFO, background="#d1ecf1")
app = pn.Column(about, info)
app.servable()
```

### Additional Context

If I then resize or refresh the window. Then it displays correctly.

![issue_markdown_w_image_markdown](https://user-images.githubusercontent.com/42288570/70017452-6ce3f800-1583-11ea-98e7-f47204b77d33.gif)

I have a feeling the Bokeh javascript some how lays out the page before actual knowing that there is a image taking up some space. But that's the feeling.

**Comment by mattpap** [comment]
This is a "well known" issue. There's a relatively simple workaround in the particular case if `<img>` elements that can be implemented (parse inner HTML, collect all `<img>` elements and set their `onload` callbacks to recompute the layout). Alternatively one can set images' dimensions in HTML, if such are known in advance. This problem is hard to solve in general with the current layout. In 3.0 we will switch to CSS-based top-level layout, so this and many other layout related issues will resolve.

**Comment by MarcSkovMadsen** [comment]
Hi @mattpap . I've tried to follow your comment above. But I simply don't know the command to retrigger.

If I use something like the below markdown and split each section into a seperate Markdown pane in Panel then it looks like

![image](https://user-images.githubusercontent.com/42288570/97536904-2d55a400-19be-11eb-8ef2-a5ab73960c76.png)


```python
# Awesome Panel ![Awesome Badge](https://cdn.rawgit.com/sindresorhus/awesome/d7305f38d29fed78fa85652e3a63e154dd8e8829/media/badge.svg)

The purpose of the Awesome Panel Project is to share knowledge on how Awesome Panel is and can become. Naturally this site is **MADE WITH PANEL**.

Awesome Panel also provides the [awesome-panel-extensions package](https://awesome-panel.readthedocs.io/en/latest/packages/awesome-panel-extensions/index.html). Checkout more awesome resources in the sidebar.

## Panel is a Powerful Framework for Analytics Apps in Python

[Panel](https://panel.holoviz.org) is very powerful framework for creating powerful analytics apps in Python **using the tools you know and love &#128013; &#10084; &#65039;**.

It's a super power for you and your team to master a framework like this if you work with **data, models and analytics**. It's great for **ML and AI** use cases and supports your process **from exploration to production and back**.

<div align="center" style="margin: 2em; max-height: 600px" >
<a href="classic-dashboard">
    <img
        alt="Classic Dashboard"
        src="https://github.com/MarcSkovMadsen/awesome-panel-assets/blob/master/videos/classic-dashboard-app.gif?raw=true"
        style="max-width:40%;max-height:300px;border-radius:2px;"
        onload="this.offsetHeight"
    >
</a><a href="detr">
    <img
        alt="DE:TR"
        src="https://github.com/MarcSkovMadsen/awesome-panel-assets/blob/master/videos/detr-app.gif?raw=true"
        style="max-width:40%;max-height:600px;border-radius:2px;"
        onload="this.offsetHeight"
    >
</a>
<a href="echarts">
    <img
        alt="ECharts"
        src="https://github.com/MarcSkovMadsen/awesome-panel-assets/blob/master/videos/echarts-app.gif?raw=true"
        style="max-width:80%;max-height:300px;border-radius:2px;"
        onload="this.offsetHeight"
    >
</a>
</div>
...
...
``

**Comment by mattpap** [comment]
This was fixed in bokeh 3.0. Except on the canvas, all components are now CSS based, so whatever happens at the CSS level, will be reflected in CSS layout. Additionally, all components have a resize observer, which allows to reevaluate computed (or canvas) layouts when size of components changes.
